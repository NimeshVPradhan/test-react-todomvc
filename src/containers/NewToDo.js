var React = require('react');

class NewToDo extends React.Component{
  constructor(props){
    super(props);
    this.state = {
      toDoItem : '',
      items : [],
      err: ''
    }

    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onChange(event){
    this.setState({
      toDoItem:event.target.value
    })
  }

  onSubmit(event){
    event.preventDefault();
    if(!this.state.toDoItem.trim()){
        this.setState({
          err: 'input field can not be emply'
        })
    }else{
    this.setState({
      err: '',
      items : [...this.state.Items, this.state.ToDoItem],
      toDoItem:''
    });
  }
  }

  render(){
    return(
      <div className="items-container">
      <div className="form-container">
      <form onSubmit ={this.onSubmit} className="form">
        <input value={this.state.ToDoItem} onChange = {this.onChange}/>
        <div className="err">{this.state.err}</div>
        <button>Submit</button>
      </form>
      </div>

      </div>
    )
  }

}

module.exports = NewToDo;
