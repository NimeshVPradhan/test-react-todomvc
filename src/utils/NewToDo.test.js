import React from 'react';
import ReactDOM from 'react-dom';
import NewToDo from '../containers/NewToDo';
import toJSON from 'enzyme-to-json';
import Enzyme, {mount, shallow, render } from 'enzyme';
// import { adapter } from 'enzyme-adapter-react-15';

const EnzymeAdapter = require('enzyme-adapter-react-16');
Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('Component: NewToDo', () => {

   it('should add an item based on the value in the state', () => {
  const component = mount(<NewToDo />);
  const preventDefault = jest.fn();
  component.setState({
    Items:['abc']
  });
  component.find('form').simulate('submit', { preventDefault });
  expect(toJSON(component)).toMatchSnapshot();
  expect(preventDefault).toBeCalled();
  });

});
