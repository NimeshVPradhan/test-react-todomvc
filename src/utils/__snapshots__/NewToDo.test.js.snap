// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NewToDo should add an item based on the value in the state 1`] = `
<NewToDo>
  <div
    className="items-container"
  >
    <div
      className="form-container"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
        />
        <div
          className="err"
        >
          input field can not be emply
        </div>
        <button>
          Submit
        </button>
      </form>
    </div>
  </div>
</NewToDo>
`;
